File: Haskell/Implementations/Decorator/Screen.hs
module Screen
    ( displayOnScreen
    ) where

import qualified BackgroundColor as B
import Color (colorCode)
import Display (Display(..))
import qualified FontColor as F

displayOnScreen :: Display a => a -> IO ()
displayOnScreen x = do
    putStrLn (display x)
    resetColors
    putStrLn ""

resetColors :: IO ()
resetColors = do
    putStr (colorCode B.defaultColor)
    putStr (colorCode F.defaultColor)

File: Java/Implementations/Decorator/Screen.java
public class Screen {
	private void resetColors() {
		System.out.print(BackgroundColor.DEFAULT.colorCode());
		System.out.print(FontColor.DEFAULT.colorCode());
	}

	public void displayOnScreen(Display d) {
		System.out.print(d.display());
		this.resetColors();
		System.out.println("");
	}
}
